src/bridge/fetch.ts

```diff
@@ -62,37 +73,38 @@ function fetchBridge(ctx: Context, config: Config, urlStr: string, init: any, bo
   })
 
 
-  setImmediate(() => {
-    req.on("response", function (res) {
-      try {
-        res.pause()
-
-        ctx.applyCallback(cb, [
-          null,
-          new ivm.ExternalCopy({
-            status: res.statusCode,
-            statusText: res.statusMessage,
-            ok: res.statusCode && res.statusCode >= 200 && res.statusCode < 400,
-            url: urlStr,
-            headers: res.headers
-          }).copyInto({ release: true }),
-          new ProxyStream(res).ref
-        ])
-
-
-      } catch (err) {
-        log.error("caught error", err)
-        ctx.tryCallback(cb, [err.toString()])
-      }
-    })
-
-    req.on("error", function (err) {
-      log.error("error requesting http resource", err)
+  req.on("response", function (res: http.IncomingMessage) {
+    log.silly(`Fetch response: ${res.statusCode} ${urlStr} ${JSON.stringify(res.headers)}`)
+    try {
+      res.pause()
+
+      ctx.applyCallback(cb, [
+        null,
+        new ivm.ExternalCopy({
+          status: res.statusCode,
+          statusText: res.statusMessage,
+          ok: res.statusCode && res.statusCode >= 200 && res.statusCode < 400,
+          url: urlStr,
+          headers: res.headers
+        }).copyInto({ release: true }),
+        new ProxyStream(res).ref
+      ])
+
+
+    } catch (err) {
+      log.error("caught error", err)
       ctx.tryCallback(cb, [err.toString()])
-    })
+    }
+  })
 
-    req.end(body && Buffer.from(body) || null)
+  req.on("error", function (err) {
+    log.error("error requesting http resource", err)
+    ctx.tryCallback(cb, [err.toString()])
   })
 
+  setImmediate(() =>
```

Comment:
not sure what the implication is of changing `setImmediate` only once request is finalized?

Reply:
It's a readability change. We used to attach the event handlers in the `setImmediate` also. It just wasn't necessary and made it harder to reason about.

We need that `setImmediate` only for when we start reading or writing to the body.